roles/azure_virtual_machine_with_public_ip/tasks/delete.yml

```diff
@@ -11,6 +11,26 @@
     remove_on_absent: "{{ azure_virtual_machine_with_public_ip_remove_on_absent | default(omit) }}"
     state: absent
 
+# Delete availability set if created by role
+- name: Get availability set info
+  azure.azcollection.azure_rm_availabilityset_info:
+    resource_group: "{{ azure_virtual_machine_with_public_ip_resource_group }}"
+    name: "{{ azure_virtual_machine_with_public_ip_availability_set.name }}"
+    tags:
+      _own_availability_set_
+  register: _aset_info
+  when: azure_virtual_machine_with_public_ip_availability_set.name is defined
```

Comment:
I think this when will need to be broken out?
when:
  - azure_virtual_machine_with_public_ip_availability_set is defined
  - azure_virtual_machine_with_public_ip_availability_set.name is defined

Reply:
name is required field, if azure_virtual_machine_with_public_ip_availability_set is defined, it should have a name.

Reply:
If you ran the test with non-existing parent it sounds like this works.

Reply:
fixed

Reply:
I'm having second thought about it, even if test passed. It is cost nothing to use 'azure_virtual_machine_with_public_ip_availability_set' as condition and it won't cause any questions.
I'll replace 'azure_virtual_machine_with_public_ip_availability_set.name' with 'azure_virtual_machine_with_public_ip_availability_set '